core/src/main/scala/org/apache/spark/sql/delta/DeltaSharedExceptions.scala

```diff
@@ -48,3 +58,17 @@ class DeltaUnsupportedOperationException(
     override def getErrorClass: String = errorClass
     def getMessageParametersArray: Array[String] = messageParameters
 }
+
+// todo: we had to add this since in Spark 3.4 ParseException(message, ...) was replaced by
+//   ParseException(errorClass, ...)
+//   Instead of passing just a message here, we could enforce creating an errorClass for each
+//   invocation and make this DeltaParseException(errorClass, ctx)
+class DeltaParseException(
```

Comment:
In Spark 3.3 there is a constructor `ParseException(message, ctx)` but in 3.4 it is `ParseException(errorClass, ctx)` (and that error class would need to be a _Spark error class_).

Here we enable creating a `ParseException` from a message and context.